rpmlint/checks/SpecCheck.py

```diff
@@ -454,23 +433,23 @@ def check_spec(self, pkg):
                 res = conflicts_regex.search(line)
                 if res:
                     confs = Pkg.parse_deps(res.group(1))
-                    e = Pkg.has_forbidden_controlchars(confs)
-                    if e:
+                    deptoken = Pkg.has_forbidden_controlchars(confs)
+                    if deptoken:
                         self.output.add_info('E', pkg,
                                              'forbidden-controlchar-found',
-                                             'Conflicts: %s' % e)
+                                             'Conflicts: %s' % deptoken)
                     for conf in unversioned(confs):
                         if compop_regex.search(conf):
                             self.output.add_info('W', pkg,
                                                  'comparison-operator-in-deptoken',
                                                  conf)
 
             if current_section == 'changelog':
-                e = Pkg.has_forbidden_controlchars(line)
-                if e:
+                deptoken = Pkg.has_forbidden_controlchars(line)
+                if deptoken:
                     self.output.add_info('E', pkg,
                                          'forbidden-controlchar-found',
-                                         '%%changelog: %s' % e)
+                                         '%%changelog: %s' % deptoken)
```

Comment:
If you notice,
all of the `forbidden-controlchar-found` are inside the single `for loop`
https://github.com/rpm-software-management/rpmlint/blob/e485305ad01f5d480ea9d9d18937ea2e32ae726e/rpmlint/checks/SpecCheck.py#L215

Since everything (checks are intertwined or depend on each other) is basically nested within this particular loop. Breaking this cycle seems not an "efficient" step as of now as it could lead to breaking of unnecessary checks. 
Although I do plan to refactor this loop properly later.

Reply:
From line https://github.com/rpm-software-management/rpmlint/blob/e485305ad01f5d480ea9d9d18937ea2e32ae726e/rpmlint/checks/SpecCheck.py#L406 to 466 those are exactly same operations just with different variable to check, thats what I ment.
You could reduce it with some dict/loop.